packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -425,108 +436,121 @@ function useMediaUploadQueue() {
 
   // Upload files to server, optionally first transcoding them.
   useEffect(() => {
-    (async () => {
-      await Promise.all(
-        /**
-         * Uploads a single pending item.
-         *
-         * @param {Object} item Queue item.
-         * @param {File} item.file File object.
-         * @return {Promise<void>}
-         */
-        state.queue.map(async (item) => {
-          const {
-            id,
-            file,
-            state: itemState,
-            resource,
-            additionalData = {},
-            muteVideo,
-            trimData,
-          } = item;
-          if (ITEM_STATUS.PENDING !== itemState) {
-            return;
-          }
+    state.queue.forEach(
+      /**
+       * Uploads a single pending item.
+       *
+       * @param {Object} item Queue item.
+       * @param {File} item.file File object.
+       * @param {Object} item.additionalData Additional Data object.
+       */
+      async (item) => {
+        const { id, file, resource, isAnimatedGif, state: itemState } = item;
+        if (ITEM_STATUS.PENDING !== itemState) {
+          return;
+        }
 
-          if (
-            resource.type === 'video' &&
-            resource.isMuted !== null &&
-            additionalData?.isMuted === undefined
-          ) {
-            additionalData.isMuted = resource.isMuted;
-          }
+        // Changing item state so that an item is never processed twice
+        // in this effect.
+        prepareItem({ id });
 
-          if (resource?.baseColor) {
-            additionalData.meta = {
-              ...additionalData.meta,
-              baseColor: resource.baseColor,
-            };
-          }
-          // Do not copy over blurhash for new trimmed videos, poster might be different.
-          if (resource?.blurHash && !resource?.trimData) {
-            additionalData.meta = {
-              ...additionalData.meta,
-              blurHash: resource.blurHash,
-            };
-          }
+        const isVideo = file.type.startsWith('video/');
 
-          const isGifThatNeedsTranscoding =
-            resource.mimeType === 'image/gif' &&
-            isAnimatedGif(await file.arrayBuffer());
+        if (isVideo) {
+          // TODO: Consider always using getFileInfo() to have more accurate audio information.
 
-          const needsTranscoding =
-            isTranscodingEnabled &&
-            (isGifThatNeedsTranscoding || canTranscodeFile(file));
+          if (
+            item.additionalData.mediaSource !== 'recording' &&
+            (await isConsideredOptimized(resource, file))
```

Comment:
For resources objects, like mpg / avi, it is a shame we can not update missing fields like mute, length etc.

Reply:
How about something like this 
```
const newResource = populateInfor(resource, file);
if( !resource.isOptimized ) {
```

Should be simple enough to change up the code as well.

Reply:
That is quite the course reversal again. IIUC last time we discussed that it can be wasteful to run MediaInfo on every resource, even if it provides useful information and that we would start with this simple approach of just checking optimization status. That's why I added the TODO following that discussion so that we could come back to this later.

Reply:
No it isn't just changing params really. As it may not be clear to you what I meant, I put together a branch. See https://github.com/GoogleForCreators/web-stories-wp/commit/843009560a5e459e0eef075a9f439437b1824b97

Basically, change to return `isConsideredOptimized`. That needs to done is update some tests and then update the media queue to use the updated resource. Could be a very simple change and makes the code more effective.

Reply:
After looking at this again, I think it might not really be worth it. The only benefit here, is adding height / width / length / isMuted to files that need to be transcoded. But these are the most useless and only displayed in the media library. It might worth adding to the code base but done in the follow up PR.

Reply:
Ah 💡 Now I understand it better! Thanks for explaining it to me, it's much appreciated.

Agreed that it would be nice to leverage the fileInfo when it's already there, but that it would also be a rather low impact change.

Let's revisit this again in the future as you suggested 👍